From eaf58af940fe6f01c1e74cc1913af2cc013baa6d Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:49:07 -0400
Subject: [message] i2o: Fix I/O space allocation copy/paste error

Message-id: <20140806034907.1438.21064.stgit@gir.stowe>
Patchwork-id: 87456
O-Subject: [RHEL7.1 PATCH 47/69] i2o: Fix I/O space allocation copy/paste error
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 7ed37fc3289d2d7d9c33abbebc0ecc9343e17101

commit 7ed37fc3289d2d7d9c33abbebc0ecc9343e17101
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Wed Feb 26 11:25:56 2014 -0700

    i2o: Fix I/O space allocation copy/paste error

    When i2o_iop_systab_set() allocates I/O port space, it assigns the base of
    the new I/O port region to sb->current_mem_base, not sb->current_io_base.
    This looks like a copy/paste error, because we do use current_io_base, but
    there's no other place that sets it.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/message/i2o/iop.c b/drivers/message/i2o/iop.c
index a8c08f3..a8373d7 100644
--- a/drivers/message/i2o/iop.c
+++ b/drivers/message/i2o/iop.c
@@ -704,7 +704,7 @@ static int i2o_iop_systab_set(struct i2o_controller *c)
            NULL, NULL) >= 0) {
    c->io_alloc = 1;
    sb->current_io_size = resource_size(res);
-   sb->current_mem_base = res->start;
+   sb->current_io_base = res->start;
    osm_info("%s: allocated %llu bytes of PCI I/O at "
     "0x%016llX.\n", c->name,
     (unsigned long long)resource_size(res),
-- 
1.7.1